libs/core-ui/src/lib/util/datasetUtils/getColumnRanges.ts

```diff
@@ -62,12 +65,20 @@ export function getColumnRanges(
 }
 
 function getIndexFeatureRange(dataset: IDataset): IColumnRange {
-  return {
-    max: dataset.features.length - 1,
-    min: 0,
-    rangeType: RangeTypes.Integer,
-    sortedUniqueValues: [...new Array(dataset.features.length).keys()]
-  };
+  return !ifEnableLargeData(dataset)
+    ? {
+        max: dataset.features.length - 1,
+        min: 0,
+        rangeType: RangeTypes.Integer,
+        sortedUniqueValues: [...new Array(dataset.features.length).keys()]
+      }
+    : ({
+        max: dataset.tabular_dataset_metadata?.num_rows
```

Comment:
nit: move this to variable
```dataset.tabular_dataset_metadata?.num_rows```

Reply:
Thanks, Ilya. I tried to have the below variable but the issue with that is, it will be initiated even when not needed (not in large data cases). If I try to have this initiated in large data cases, it adds more lines of code. Open to change if this can be done in an easier way :)
_const numberOfRows = dataset.tabular_dataset_metadata?.num_rows;_